examples/kubernetesconfiguration/v1beta1/kubernetesclusterextension.yaml

```
@@ -0,0 +1,54 @@
+apiVersion: kubernetesconfiguration.azure.upbound.io/v1beta1
+kind: KubernetesClusterExtension
+metadata:
+  annotations:
```

> But then there is not enough time for the extension to get created:
logger.go:42: 20:34:01 | case/0-apply | test step failed 0-apply

We have `uptest.upbound.io/timeout` annotation for such cases. Could you please add and re-trigger the uptest?

```suggestion
  annotations:
    uptest.upbound.io/timeout: "3600" # 60 min timeout
```

---

It seems the cluster creation doesn't succeed:
 16:55:32      reason: Creating 16:55:32      status: "False" 16:55:32      type: Ready

---

Without the new extension feature in the example CR the exact same AKS cluster as it was defined before was created successfully. This is [the commit](https://github.com/crossplane-contrib/provider-upjet-azure/pull/805/commits/bfb2f960f0454a77d1efd2404bbfab74e10c5e18).
This could suggest that tapping into {{ cluster_id }}/providers/Microsoft.KubernetesConfiguration/extensions/ while the cluster is in creating state leads to a race condition and the cluster could get stuck in creating state because there is a diff in the configuration - and the extension CR would never get the clusterID from the reference because it is not in the Clusters CR before it is done creating. I cannot prove this for now but this is my best guess.
Would be nice to test this by creating the cluster first and then trying to tap in to the extensions endpoint with the CR.

---

@turkenf sorry for the information overflow, just trying to make sense of what is happening. In summary, is there a way to tell the CI to create the AKS cluster first and then create the CR for the extension?

---

[This run,](https://github.com/crossplane-contrib/provider-upjet-azure/actions/runs/10673411111/job/29582428830) the cluster created successfully and the extension CR got the clusterId using the reference. It is in creating state but for way too long. Would be great to see some output from Azure API side, might be some misconfig with the extension.

---

I think I understand what is going on. In my own env when I apply the aks cluster first and wait until it's created before applying the extension then it works. If I apply the CR in one go, same as the uptests, the extension never gets created and is stuck in creating state. @turkenf is there any way to handle these cases? Essentially we would need the AKS cluster to be ready first and then apply the extension.

---

The above is incorrect - the extension should not interfere with the cluster until it gets the clusterId from the ref selector. I have no explanation on why it works on my env according to the manual tests and not with the pipeline.

---

This does not seem to be the root cause - [opened an issue.](https://github.com/crossplane-contrib/provider-upjet-azure/issues/809)